app/controllers/api/v4/rest/CacheController.scala

```diff
@@ -0,0 +1,36 @@
+package controllers.api.v4.rest
+
+import com.typesafe.config.Config
+import javax.inject.Inject
+import models.gql.Fetchers
+import play.api.Logging
+import play.api.mvc._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CacheController @Inject()(implicit ec: ExecutionContext,
+                                cc: ControllerComponents, config: Config)
+  extends AbstractController(cc)
+    with Logging {
+
+  def clearCache(): Action[AnyContent] = checkCredentials(Action.async { implicit request =>
+    Future {
+      logger.info("Received request to clear cache.")
+      Fetchers.resetCache()
+      Ok("Cache cleared.")
+    }
+  })
+
+  def checkCredentials[A](action: Action[A]): Action[A] = Action.async(action.parser) { request =>
```

Comment:
@JarrodBaker it is great you used action compositions here. more specifically the easy method wrapping one. It would be good if you can take this method from here and put it in a common helper place for controllers as it can be composed with any other action really.